tests/Feature/Repositories/CartTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace IndieHD\Velkart\Tests\Feature\Repositories;
+
+use IndieHD\Velkart\Contracts\OrderRepositoryContract;
+use IndieHD\Velkart\Contracts\CartRepositoryContract;
+use Ramsey\Uuid\UuidFactoryInterface;
+
+class CartTest extends RepositoryTestCase
+{
+    /*
+     * @var UuidFactoryInterface
+     */
+    protected $uuid;
+
+    /*
+     * @var OrderRepositoryContract
+     */
+    protected $order;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setRepository(resolve(CartRepositoryContract::class));
+
+        $this->uuid = resolve(UuidFactoryInterface::class);
+
+        $this->order = resolve(OrderRepositoryContract::class);
+    }
+
+    /** @test */
+    public function itCanUpdate()
+    {
+        $cart = factory($this->getRepository()->modelClass())->create();
+
+        $identifier = $this->uuid->uuid4();
+
+        $updates = [
+            'identifier' => $identifier->toString()
+        ];
+
+        $updated = $this->getRepository()->update($cart->id, $updates);
+
+        $this->assertTrue($updated);
+        $this->assertDatabaseHas($this->getRepository()->model()->getTable(), $updates);
+    }
+
+    /** @test */
+    public function itHasOneOrder()
+    {
+        $order = factory($this->order->modelClass())->create();
+
+        $cart = $this->getRepository()->findById($order->cart->id);
```

Comment:
This approach is not intuitive. I was being lazy and didn't rig-up the `Cart` factory to have a `withOrder` state, so this test is relying on the `Order` factory (which creates both related models). So, while it's "accurate", it's confusing. I should rework this.